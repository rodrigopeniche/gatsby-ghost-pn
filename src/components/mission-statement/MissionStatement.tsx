import * as React from "react"
import MissionStatementContainer from "./MissionStatementContainer"
import TextArea from "./TextArea"
import Inputs from "./Inputs"
import SearchButton from "./Button"

const MissionStatement = () => {
    return (
        <MissionStatementContainer>
            <TextArea />
            <Inputs />
            <SearchButton />
        </MissionStatementContainer>
    )
}

export default MissionStatement
