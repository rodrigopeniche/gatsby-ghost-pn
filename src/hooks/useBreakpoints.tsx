import * as React from "react"
import { useState, useEffect, createContext, useContext } from "react"

const defaultValue = {}

const BreakpointContext = createContext(defaultValue)

const BreakpointProvider = ({
    children,
    queries,
}: {
    children: React.ReactNode | React.ReactNode[]
    queries: { [key: string]: string }
}) => {
    const [queryMatch, setQueryMatch] = useState({})

    useEffect(() => {
        const mediaQueryLists: { [key: string]: any } = {}
        const keys = Object.keys(queries)
        let isAttached = false

        const handleQueryListener = () => {
            const updatedMatches = keys.reduce(
                (acc: { [key: string]: any }, media) => {
                    acc[media] = !!(
                        mediaQueryLists[media] && mediaQueryLists[media].matches
                    )
                    return acc
                },
                {}
            )
            setQueryMatch(updatedMatches)
        }

        if (window && window.matchMedia) {
            const matches: { [key: string]: any } = {}
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media] = window.matchMedia(queries[media])
                    matches[media] = mediaQueryLists[media].matches
                } else {
                    matches[media] = false
                }
            })
            setQueryMatch(matches)
            isAttached = true
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media].addListener(handleQueryListener)
                }
            })
        }

        return () => {
            if (isAttached) {
                keys.forEach(media => {
                    if (typeof queries[media] === "string") {
                        mediaQueryLists[media].removeListener(
                            handleQueryListener
                        )
                    }
                })
            }
        }
    }, [queries])

    return (
        <BreakpointContext.Provider value={queryMatch}>
            {children}
        </BreakpointContext.Provider>
    )
}

function useBreakpoint(): { [key: string]: boolean } {
    const context = useContext(BreakpointContext)
    if (context === defaultValue) {
        throw new Error("useBreakpoint must be used within BreakpointProvider")
    }
    return context
}
export { useBreakpoint, BreakpointProvider }
