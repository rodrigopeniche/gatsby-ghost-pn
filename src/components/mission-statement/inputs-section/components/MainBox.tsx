import * as React from "react"
import "./RemoveArrows.css"
import { useBreakpoint } from "../../../../hooks/useBreakpoints"

const MainBox = ({
    name,
    value,
    inputHandler,
}: {
    name: string
    value: string
    inputHandler(name: string, value: string): void
}) => {
    const { md } = useBreakpoint()

    const mainBoxesStyles: React.CSSProperties = {
        maxWidth: "70px",
        minWidth: "50px",
        boxSizing: "border-box",
        height: "33px",
        marginTop: "0px",
        textAlign: "center",
        background: "#FFFAC1",
        border: "none",
        borderRadius: "5px",
        marginRight: "2px",
        marginLeft: "2px",
        marginBottom: "0px",
        display: "inline-block",
        ...(md && { marginRight: "4px", marginLeft: "4px" }),
    }

    return (
        <input
            type="number"
            name={name}
            size={4}
            value={value?.substr(0, 4) || ""}
            style={mainBoxesStyles}
            onChange={e => inputHandler(name, e.target.value)}
            required
        />
    )
}

export default MainBox
