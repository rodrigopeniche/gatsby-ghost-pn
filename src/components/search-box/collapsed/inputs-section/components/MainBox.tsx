import * as React from "react"
import "./RemoveArrows.css"
import { useBreakpoint } from "../../../../../hooks/useBreakpoints"

const MainBox = ({
    name,
    value,
    inputHandler,
    setHover,
}: {
    name: string
    value: string
    inputHandler(name: string, value: string): void
    setHover: Function
}) => {
    const { md } = useBreakpoint()

    const mainBoxesStyles: React.CSSProperties = {
        maxWidth: "70px",
        minWidth: "50px",
        boxSizing: "border-box",
        height: "33px",
        marginTop: "0px",
        background: "#FFFAC1",
        border: "none",
        borderRadius: "5px",
        marginRight: "8px",
        marginLeft: "8px",
        marginBottom: "0px",
        display: "inline-block",
        textAlign: "center",
        ...(md && { marginRight: "4px", marginLeft: "4px" }),
    }

    return (
        <input
            type="number"
            name={name}
            size={3}
            value={value?.substr(0, 3) || ""}
            style={mainBoxesStyles}
            onChange={e => inputHandler(name, e.target.value)}
            onMouseEnter={() => setHover(true)}
            onMouseLeave={() => setHover(false)}
            disabled
            required
        />
    )
}

export default MainBox
