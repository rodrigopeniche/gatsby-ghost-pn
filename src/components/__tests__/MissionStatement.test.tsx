import * as React from "react"
import * as renderer from "react-test-renderer"
import "@testing-library/jest-dom/extend-expect"
import { BreakpointProvider } from "../../hooks/useBreakpoints"
import MissionStatement from "../mission-statement/MissionStatement"
import MainBox from "../mission-statement/inputs-section/components/MainBox"

const queries = {
    md: "(max-width: 767px)",
    sm: "(max-width: 600px)",
    xs: "(max-width: 480px)",
    ms: "(max-width: 420px)",
    ss: "(max-width: 320px)",
}

describe("MissionStatement", () => {
    const searchBox = (
        <BreakpointProvider queries={queries}>
            <MissionStatement />
        </BreakpointProvider>
    )

    const component = renderer.create(searchBox)

    it("renders correctly", () => {
        const jsonTree = component.toJSON()
        expect(jsonTree).toMatchSnapshot()
    })

    it("have number inputs", () => {
        const boxes = component.root.findAllByType(MainBox)
        expect(boxes.length).toBe(3)
    })
})
