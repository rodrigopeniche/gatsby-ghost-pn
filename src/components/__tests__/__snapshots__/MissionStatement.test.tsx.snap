// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionStatement renders correctly 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundImage": "url(undefined)",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "borderRadius": "5px",
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginLeft": "11px",
      "marginRight": "11px",
      "maxWidth": "929px",
      "minHeight": "187px",
      "minWidth": "250px",
      "textAlign": "left",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "marginLeft": "24px",
        "marginTop": "5px",
      }
    }
  >
    <h1
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Georgia",
          "fontSize": "36px",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "justifyContent": "normal",
          "lineHeight": "41px",
          "marginBottom": "0",
          "marginTop": "0",
        }
      }
    >
      Mission statement
    </h1>
    <p
      style={
        Object {
          "color": "#FFFFFF",
          "display": "flex",
          "fontFamily": "Georgia",
          "fontSize": "19px",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "justifyContent": "normal",
          "lineHeight": "34px",
          "margin": "0",
          "textAlign": "center",
        }
      }
    >
      is placeholder text commonly used in the graphic
    </p>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexWrap": "wrap",
        "fontFamily": "Arial",
        "fontStyle": "normal",
        "fontWeight": "normal",
        "height": "60px",
        "justifyContent": "center",
        "marginRight": "-50px",
        "marginTop": "13px",
        "position": "relative",
        "textAlign": "center",
        "width": "262px",
      }
    }
  >
    <p
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "#FFFFFF",
          "display": "inline-block",
          "fontSize": "30px",
          "lineHeight": "34px",
          "marginBottom": "0px",
          "paddingBottom": "2.5px",
          "verticalAlign": "middle",
        }
      }
    >
      1
    </p>
    <div
      style={
        Object {
          "display": "flex",
          "flexBasis": "78px",
          "flexGrow": 1,
          "marginRight": "5px",
          "maxWidth": "98px",
          "minWidth": "78px",
        }
      }
    >
      <p
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#FFFFFF",
            "fontFamily": "Arial",
            "fontSize": "30px",
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": "34px",
            "marginBottom": "0px",
            "paddingBottom": "5px",
          }
        }
      >
        (
      </p>
      <div
        style={
          Object {
            "display": "flex",
            "flexBasis": "58px",
            "flexDirection": "column",
            "flexGrow": 1,
            "maxWidth": "78px",
            "minWidth": "58px",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <label
          style={
            Object {
              "color": "#FFFFFF",
              "display": "block",
              "fontFamily": "Arial",
              "fontSize": "12px",
              "fontStyle": "normal",
              "lineHeight": "14px",
              "marginBottom": "12px",
              "textAlign": "center",
            }
          }
        >
          Area code
        </label>
        <input
          name="area-code"
          onChange={[Function]}
          required={true}
          size={4}
          style={
            Object {
              "background": "#FFFAC1",
              "border": "none",
              "borderRadius": "5px",
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "33px",
              "marginBottom": "0px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "marginTop": "0px",
              "maxWidth": "70px",
              "minWidth": "50px",
              "textAlign": "center",
            }
          }
          type="number"
          value=""
        />
      </div>
      <p
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#FFFFFF",
            "fontFamily": "Arial",
            "fontSize": "30px",
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": "34px",
            "marginBottom": "0px",
            "paddingBottom": "5px",
          }
        }
      >
        )
      </p>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexBasis": "58px",
          "flexDirection": "column",
          "flexGrow": 1,
          "maxWidth": "63px",
          "minWidth": "58px",
          "position": "relative",
        }
      }
    >
      <label
        style={
          Object {
            "color": "#FFFFFF",
            "display": "block",
            "fontFamily": "Arial",
            "fontSize": "12px",
            "fontStyle": "normal",
            "lineHeight": "14px",
            "marginBottom": "12px",
            "textAlign": "center",
          }
        }
      >
        Exchange
      </label>
      <input
        name="exchange"
        onChange={[Function]}
        required={true}
        size={4}
        style={
          Object {
            "background": "#FFFAC1",
            "border": "none",
            "borderRadius": "5px",
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "33px",
            "marginBottom": "0px",
            "marginLeft": "2px",
            "marginRight": "2px",
            "marginTop": "0px",
            "maxWidth": "70px",
            "minWidth": "50px",
            "textAlign": "center",
          }
        }
        type="number"
        value=""
      />
    </div>
    <div
      style={
        Object {
          "alignSelf": "flex-end",
          "borderTop": "2px solid #FFFFFF",
          "marginBottom": "18px",
          "marginLeft": "7px",
          "marginRight": "7px",
          "width": "17px",
        }
      }
    />
    <div
      style={
        Object {
          "display": "flex",
          "flexBasis": "58px",
          "flexDirection": "column",
          "flexGrow": 1,
          "maxWidth": "63px",
          "minWidth": "58px",
          "position": "relative",
        }
      }
    >
      <label
        style={
          Object {
            "color": "#FFFFFF",
            "display": "block",
            "fontFamily": "Arial",
            "fontSize": "12px",
            "fontStyle": "normal",
            "lineHeight": "14px",
            "marginBottom": "12px",
            "textAlign": "center",
          }
        }
      >
        Subscriber
      </label>
      <input
        name="subscribe"
        onChange={[Function]}
        required={true}
        size={4}
        style={
          Object {
            "background": "#FFFAC1",
            "border": "none",
            "borderRadius": "5px",
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "33px",
            "marginBottom": "0px",
            "marginLeft": "2px",
            "marginRight": "2px",
            "marginTop": "0px",
            "maxWidth": "70px",
            "minWidth": "50px",
            "textAlign": "center",
          }
        }
        type="number"
        value=""
      />
    </div>
  </div>
  <div
    style={
      Object {
        "display": "block",
        "margin": "39px 0 0 0",
        "textAlign": "center",
        "width": "113px",
      }
    }
  >
    <div
      style={
        Object {
          "background": "linear-gradient(180deg, #FD1D00 0%, #B81200 100%)",
          "borderRadius": "5px",
          "display": "inline-block",
          "filter": "drop-shadow(0px 1px 8px rgba(0, 0, 0, 0.6))",
          "float": "right",
          "height": "34px",
          "marginRight": "21px",
          "width": "100%",
        }
      }
    >
      <h1
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Georgia",
            "fontSize": "15px",
            "fontWeight": "bold",
            "justifyContent": "center",
            "lineHeight": "37px",
            "marginBottom": "0",
            "textAlign": "center",
          }
        }
      >
        SEARCH
      </h1>
    </div>
  </div>
</div>
`;
